git_pull/run.sh

```diff
@@ -224,17 +228,26 @@ function validate-config {
 #### Main program ####
 cd /config || { echo "[Error] Failed to cd into /config"; exit 1; }
 
+if [ "$WEBHOOK_ACTIVE" == "true" -a "$WEBHOOK_RUNNING" != "true" ]; then
+    webhook -verbose -template -port 8004 -hooks /hook-template.json &
+    WEBHOOK_PID=$!
+fi
+
 while true; do
     check-ssh-key
     setup-user-password
     if git-synchronize ; then
         validate-config
     fi
-     # do we repeat?
-    if [ ! "$REPEAT_ACTIVE" == "true" ]; then
+    # do we repeat?
+    if [ "$REPEAT_ACTIVE" == "true" ]; then
+        sleep "$REPEAT_INTERVAL"
+    # or do we just wait for incoming webhooks?
+    elif [ "$WEBHOOK_ACTIVE" == "true" -a "$WEBHOOK_RUNNING" != "true" ]; then
```

Comment:
I would personally leave this loop alone and not change the default behavior. Webhooks can be a nice addition, but relying upon them sololy seems like a bad idea to me.

WebHooks can be missed, or not delivered (Especially GitLab is really good at this). Still having the normal cycle would be nice.

Reply:
Hi @frenck,

I think we still have the loop here as long as `$REPEAT_ACTIVE` is still enabled.
If its disabled, we will wait for the webhook server to terminate.

The `$WEBHOOK_RUNNING` env variable is set for invocations of the script which are triggered by webhooks.

The normal behaviour of the script isn't touched.

Reply:
👍